lex_test.go

```diff
@@ -0,0 +1,275 @@
+package mql
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_lexKeywordState(t *testing.T) {
```

Comment:
happy to.

Reply:
btw, this was sort of intentional, because all states transition back to sort of "steady state" of `lexStartState` so it has to handle every possibility.